source/FreeRTOS_IP.c

```diff
@@ -1797,23 +1797,31 @@ static eFrameProcessingResult_t prvProcessIPPacket( const IPPacket_t * pxIPPacke
     {
         #if ( ipconfigUSE_IPv6 != 0 )
             case ipIPv6_FRAME_TYPE:
-                /* MISRA Ref 11.3.1 [Misaligned access] */
-                /* More details at: https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/MISRA.md#rule-113 */
-                /* coverity[misra_c_2012_rule_11_3_violation] */
-                pxIPHeader_IPv6 = ( ( const IPHeader_IPv6_t * ) &( pxNetworkBuffer->pucEthernetBuffer[ ipSIZE_OF_ETH_HEADER ] ) );
-
-
-                uxHeaderLength = ipSIZE_OF_IPv6_HEADER;
-                ucProtocol = pxIPHeader_IPv6->ucNextHeader;
-                /* MISRA Ref 11.3.1 [Misaligned access] */
-                /* More details at: https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/MISRA.md#rule-113 */
-                /* coverity[misra_c_2012_rule_11_3_violation] */
-                eReturn = prvAllowIPPacketIPv6( ( ( const IPHeader_IPv6_t * ) &( pxIPPacket->xIPHeader ) ), pxNetworkBuffer, uxHeaderLength );
-
-                /* The IP-header type is copied to a location 6 bytes before the messages
-                 * starts.  It might be needed later on when a UDP-payload buffer is being
-                 * used. */
-                pxNetworkBuffer->pucEthernetBuffer[ 0 - ( BaseType_t ) ipIP_TYPE_OFFSET ] = pxIPHeader_IPv6->ucVersionTrafficClass;
+
+                if( pxNetworkBuffer->xDataLength < sizeof( IPPacket_IPv6_t ) )
```

Comment:
Is it possible to have frame type of ipIPv6_FRAME_TYPE and length less than IPPacket_IPv6_t? Why is this check being added now?

Reply:
Before calling `prvProcessIPPacket`, we only check if its size is greater than IPv4 packet in [prvProcessEthernetPacket](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/dev/IPv6_integration/source/FreeRTOS_IP.c#L1585). But it's not enough for IPv6. The condition here to make sure the buffer size is also valid for IPv6 packets.